placeSpecificContent/hq/courtyard/controlPanel/controlPanel_server.js

```diff
@@ -0,0 +1,406 @@
+        mode requested. */
+        playSound: function(sound, volume, position, localOnly, loop) {
+            if (sound.downloaded) {
+                if (injector) {
+                    injector.stop();
+                    injector = null;
+                }
+                injector = Audio.playSound(sound, {
+                    position: position,
+                    volume: volume,
+                    localOnly: localOnly,
+                    loop: loop
+                });
+            }
+        },
+
+        /* Remotely called when the user presses a button. The id of the button is sent as a param and then used to 
+        match an index of the map to get target positions for each entity. Call another function to set up movement 
+        for each entity and then set the current button to transition color and all other buttons to inactive color.
+         Play the transition sound and edit the velocity of entities that will move.
+         */
+        pressButton: function(thisID, params) {
+            if (!ready) {
+                return;
+            }
+
+            courtyardInPlace = false;
+            roundtableInPlace = false;
+            courtyardVelocity = 0;
+            roundtableVelocity = 0;
+
+            var buttonID = params[0];
+            
+            var targetCourtyardPosition = BUTTON_TO_ENTITY_POSITIONING_MAP[buttonID].courtyardPosition;
+            var targetRoundtablePosition = BUTTON_TO_ENTITY_POSITIONING_MAP[buttonID].roundtablePosition;
+
+            _this.setUpMovement(roundtable, targetRoundtablePosition);
+            _this.setUpMovement(courtyard, targetCourtyardPosition);
+
+            Object.keys(BUTTON_TO_ENTITY_POSITIONING_MAP).forEach(function(button) {
+                if (button === buttonID) {
+                    Entities.editEntity(button, {
+                        backgroundColor: BUTTON_TRANSITION_COLOR,
+                        textColor: BUTTON_INACTIVE_COLOR
+                    });
+                } else {
+                    Entities.editEntity(button, {
+                        backgroundColor: BUTTON_INACTIVE_COLOR,
+                        textColor: BUTTON_TRANSITION_COLOR
+                    });
+                }
+            });
+
+            var soundPosition = Entities.getEntityProperties(courtyard, 'position').position;
+            _this.playSound(TRANSITION_SOUND, TRANSITION_VOLUME, soundPosition, false, true);
+            
+            if (!roundtableInPlace && roundtableVelocity) {
+                Entities.editEntity(roundtable, {
+                    velocity: { x: 0, y: roundtableVelocity, z: 0 }
+                });
+            }
+
+            if (!courtyardInPlace && courtyardVelocity) {
+                Entities.editEntity(courtyard, {
+                    velocity: { x: 0, y: courtyardVelocity, z: 0 }
+                });
+            }
+
+            if (checkInterval) {
```

Comment:
why wouldn't this be the first thing you do inside `pressButton`?

Reply:
I think it should be moved up to the top of the function but still after the check for ready so we avoid an extra check if the script was unable to collect the data needed to handle state changes.